jme3-lwjgl3/src/main/java/com/jme3/system/lwjgl/LwjglWindow.java

```diff
@@ -477,10 +479,21 @@ public void create(boolean waitFor) {
             return;
         }
 
-        // NOTE: this is required for Mac OS X!
-        mainThread = Thread.currentThread();
-        mainThread.setName("jME3 Main");
-        run();
+        if (Platform.get() == Platform.MACOSX) {
+            // NOTE: this is required for Mac OS X!
+            mainThread = Thread.currentThread();
+            mainThread.setName("jME3 Main");
+            if (waitFor) {
+                LOGGER.warning("create(true) is not supported for macOS!");
+            }
+            run();
+        } else {
+            new Thread(this, "jME3 Main").start();
```

Comment:
A bit late to the party!

@bob0bob `mainThread` should be initialized to newly create thread inside `else` statement.